src/components/Table.js

```diff
@@ -1,80 +1,104 @@
-import React, { useMemo, useState } from "react";
-import trimmed_keypoints from "../data/trimmed_keypoints.json";
+import React, { useCallback, useMemo, useState } from "react";
+import trimmedKeypoints from "../data/trimmed_keypoints.json";
 import Filter from "./Filter";
 import Sorting from "./Sorting";
 import DisplaySetting from "./DisplaySetting";
 import Pagination from "./Pagination";
-import Paginate, { paginate } from "./Paginate";
+import paginate from "../utils/Paginate";
+
 function Table() {
-  // You can use below imageMap to poppulate data, it wont contain any duplicate images
-  // const [filteredImages]
-  // write a function , get the result
-  // later named anything handleFilteredResult should set the filter state then use the useMemo to get the filtered images
   const images = useMemo(() => {
     const imageMap = new Map();
-    trimmed_keypoints.images.forEach((item) => {
+    trimmedKeypoints.images.forEach((item) => {
       imageMap.set(item.id, item);
     });
     return imageMap;
-  }, [trimmed_keypoints.images]); // showing warning
-  //    Line 12:6:  React Hook useMemo has an unnecessary dependency: 'trimmed_keypoints.images'. Either exclude it or remove the dependency array.
-  //    Outer scope values like 'trimmed_keypoints.images' aren't valid dependencies because mutating them doesn't re-render the component
-  //    react-hooks/exhaustive-deps
+  }, []);
   const [attributesName] = useState(Object.keys(images.values().next().value));
   const [showFilter, setShowFilter] = useState(false);
   const [showSorting, setShowSorting] = useState(false);
   const [showDisplaySetting, setShowDisplaySetting] = useState(false);
   const [currentPage, setCurrentPage] = useState(1);
-  const [pageSize] = useState(5);
+  const [pageSize] = useState(4);
+  const [outputImages, setOutputImages] = useState(Array.from(images.values()));
   const [outputImagesAttributes, setOutputImagesAttributes] = useState(
     Object.keys(images.values().next().value)
   );
-
-  console.log(attributesName);
   function handleFilterClick() {
     setShowFilter(true);
-    // setShowSorting(false);
-    // setShowDisplaySetting(false);
   }
   function handleSortingClick() {
-    // setShowFilter(false);
     setShowSorting(true);
-    // setShowDisplaySetting(false);
   }
   function handleDisplaySettingClick() {
-    // setShowFilter(false);
-    // setShowSorting(false);
     setShowDisplaySetting(true);
   }
   function handleOutputImagesAttributes(attributesToShow) {
-    console.log("came in handle output images attributes");
     setOutputImagesAttributes(attributesToShow);
   }
   function handlePageChange(page) {
     setCurrentPage(page);
-    console.log("in handlePageChange page= ", page);
   }
-  const pageImages = paginate(
-    Array.from(images.values()),
-    currentPage,
-    pageSize
+  const pageImages = paginate(outputImages, currentPage, pageSize);
+  const performSorting = useCallback(
```

Comment:
Earlier I was using simple function but it was not working, then I tried to use callback , I thought that it was not working beacause of asynchronous nature